packages/react-charts/src/components/ChartLabel/ChartLabel.tsx

```diff
@@ -246,7 +245,7 @@ export const ChartLabel: React.FunctionComponent<ChartLabelProps> = ({
     );
   const newStyle = Array.isArray(style) ? style.map(applyDefaultStyle) : applyDefaultStyle(style);
 
-  return <VictoryLabel style={newStyle as any} textAnchor={textAnchor} {...rest} id={id} />;
```

Comment:
Was the issue with failing test, the the id provided by consumer was being overwritten, or the fact that it was generated and not predictable.  If it is the former, then spreading the `{...rest}` props after the id would fix that.

Reply:
It was the unpredictability - trying to test svg imgs on charts as a way to make sure their data is displayed correctly.. but part of those svgs included the ids, so when they are generated hashes (unpredictable) it was failing the snaps

Reply:
it was the fact that it was generated and not predictable.
And by the time the ChartLabel is rendered, many more props are added to the `...rest` including a hardcoded `id`. That's why I add the id prop provided by the consumer after the rest props are spread.

Reply:
Seems odd that you would need to do that because id has already been destructured -- the rest prop should not include it?